src/main/java/edu/ithaca/dturnbull/bank/BankAccount.java

```diff
@@ -9,15 +9,89 @@
      * @throws IllegalArgumentException if email is invalid
      */
     public BankAccount(String email, double startingBalance){
-        if (isEmailValid(email)){
-            this.email = email;
-            this.balance = startingBalance;
-        }
-        else {
+        if (!isEmailValid(email)){
             throw new IllegalArgumentException("Email address: " + email + " is invalid, cannot create account");
         }
+
+        //CODE REVIEW EDIT - included a check for starting balance validity
+        if(!isAmountValid(startingBalance)){
+            throw new IllegalArgumentException("Starting balance is invalid, cannot create account");
+        }
+
+        this.balance = startingBalance;
+        this.email = email;
+    }
+
+    /**
+     * 
+     * @param amount
+     */
+
+    public static boolean isAmountValid(double amount){
+        //negaitve
+        if(amount<0){
+            return false;
+        }
+
+        //0
+        if(amount==0){
+            return true;
+        }
+        
+        //converts it to a string and counts the number of numbers after the decimal
+        String str=Double.toString(amount);
+        for(int i=0; i<str.length();i++){
+            if(str.contains(".")){
+                String decimal=str.substring(str.lastIndexOf(".")+1);
+                double dvalues=decimal.length();
+                if(dvalues<=2){
+                    return true;
+                }
+                return false;
+            }
+            
+        }
+        
+        return true;
+    }
+
+    /**
+     * Deposits money into the bank account
+     * @param amount
+     * @throws IllegalArgumentException
+     */
+    public void deposit(double amount){
+        //checks if amount is valid
+        if(isAmountValid(amount)){
+            balance += amount;
+        }
+        else{
+            throw new IllegalArgumentException("Invalid Deposit Amount");
+        }
     }
 
+    public static void transfer(BankAccount account1, BankAccount account2, double amount) throws InsufficientFundsException{
```

Comment:
Should probably take an OOP approach and make this a (non static) function that takes only one BankAccount as an input parameter "otherAccount".